doc/whats_new.rst

```diff
@@ -171,6 +171,16 @@ Enhancements
      removed by setting it to `None`.
      :issue:`7674` by:user:`Yichuan Liu <yl565>`.
 
+   - :class:`preprocessing.OneHotEncoder` now fits and transforms inputs of
```

Comment:
I would split this between enhancements (the new stuff) and api changes (deprecations).